subprojects/gst-plugins-good/gst/rtp/gstrtph266pay.c

```diff
@@ -0,0 +1,792 @@
+/* GStreamer
+ * Copyright (C) <2024> Carlos Falgueras García <cfalgueras@fluendo.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+/* references:
+ * - RTP Payload Format for Versatile Video Coding (VVC)
+ *   https://www.ietf.org/rfc/rfc9328.txt
+ * - RTP: A Transport Protocol for Real-Time Applications
+ *   https://www.ietf.org/rfc/rfc3550.txt
+ * - H.266: Versatile video coding
+ *   https://www.itu.int/rec/T-REC-H.266-202309-I
+ */
+
+#include "gstrtph266pay.h"
+#include <gst/rtp/gstrtppayloads.h>
+#include <gst/rtp/gstrtpelements.h>
+#include <gst/rtp/gstrtpbuffer.h>
+#include <gst/rtp/gstrtputils.h>
+#include <gst/base/gstadapter.h>
+
+#define GST_CAT_DEFAULT rtph266pay_debug
+GST_DEBUG_CATEGORY_STATIC (GST_CAT_DEFAULT);
+
+typedef enum
+{
+  NALU_TYPE_VPS = 14,
+  NALU_TYPE_SPS = 15,
+  NALU_TYPE_PPS = 16,
+  NALU_TYPE_PAPS = 17,
+  NALU_TYPE_SAPS = 18,
+  NALU_TYPE_AUD = 20,
+} NaluType;
+
+#define NALU_SC_MSK 0xffffff00
+#define NALU_SC_VAL 0x00000100
+#define NALU_SC_LEN 3
+#define NALU_HDR_LEN 2
+#define NALU_INVALID_XPS 0xFF
+#define NALU_IS_PARAMETER_SET(nalu) \
+  (((nalu)->type >= NALU_TYPE_VPS) && ((nalu)->type <= NALU_TYPE_SAPS))
+
+#define FU_TYPE 29
+#define FU_HDR_LEN 3            // PayloadHdr + FU header
+
+typedef struct
+{
+  GstBuffer *nalu_buf;
+  GstBuffer *hdr_buf;
+  GstBuffer *rbsp_buf;
+  guint16 size;                 // Size of the NALU (inluding its header)
+  NaluType type;                // nal_unit_type
+  guint8 xps_id;                // {vps_video,sps_seq,pps_pic,aps_adaptation}_parameter_set_id
+  gboolean au_start;
+  gboolean au_end;
+} Nalu;
+
+#define NALU_PTR_FORMAT \
+  "p, size: %u, type: %u, xps_id: %u, au_start: %s, au_end: %s"
+#define NALU_ARGS(nalu) \
+  (nalu), (nalu)->size, (nalu)->type, (nalu)->xps_id, (nalu)->au_start ? "true" : "false", (nalu)->au_end ? "true" : "false"
+
+typedef enum
+{
+  ALIGNMENT_AU,
```

Comment:
I would suggest to prefix with `GST_` as in https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/gst-plugins-good/gst/rtp/gstrtph265pay.h#L45

Reply:
I have my doubts regarding how to address this prefix topic...
* https://github.com/fluendo/flu-plugins/pull/632#discussion_r1839675041
* https://fluendo.atlassian.net/wiki/spaces/ENG/pages/627802115/Coding+Guidelines?focusedCommentId=4321607686

Reply:
I agree